mobile/lib/widgets/chart.dart

```diff
@@ -0,0 +1,396 @@
+
+  /// A subset of [widget.series] of size [_condensedRowCount].
+  List<Series<T>> _displayData = [];
+
+  int _maxRowCount = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    _reset();
+  }
+
+  @override
+  void didUpdateWidget(Chart<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.series != widget.series) {
+      _reset();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        _buildLegend(),
+        _buildChart(),
+        _buildViewAll(),
+      ],
+    );
+  }
+
+  Widget _buildLegend() {
+    if (widget.series.length <= 1) {
+      return Empty();
+    }
+    return Padding(
+      padding: widget.padding.copyWith(
+        top: paddingWidgetSmall,
+        bottom: paddingWidgetSmall,
+      ),
+      child: Wrap(
+        spacing: paddingWidget,
+        runSpacing: paddingWidgetTiny,
+        children: widget.series.map((series) => Row(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Container(
+              width: _legendIndicatorSize,
+              height: _legendIndicatorSize,
+              color: series._color,
+            ),
+            HorizontalSpace(paddingWidgetSmall),
+            Text(series.displayDateRange.title(context)),
+          ],
+        )).toList(),
+      ),
+    );
+  }
+
+  Widget _buildChart() {
+    List<Widget> children = [];
+
+    // Get largest column value.
+    double maxValue = 0;
+    for (Series series in _displayData) {
+      int max = series.maxValue;
+      if (max > maxValue) {
+        maxValue = max.toDouble();
+      }
+    }
+
+    double maxWidth = MediaQuery.of(context).size.width;
+
+    // For every unique NamedEntity in the series, create a child widget.
+    List<T> entities = _displayData.first.data.keys.toList();
+    for (T entity in entities) {
+      for (Series series in _displayData) {
+        children.add(_buildChartRow(maxWidth, maxValue, entity, series,
+            series.data[entity],
+            series._color ?? Theme.of(context).primaryColor));
+
+        // Add space between series rows.
+        children.add(VerticalSpace(series == _displayData.last
+            ? 0 : paddingWidgetTiny));
+      }
+
+      // Add space between rows.
+      children.add(VerticalSpace(entity == entities.last ? 0 : paddingDefault));
+    }
+
+    return Container(
+      padding: widget.padding,
+      width: maxWidth,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: children,
+      ),
+    );
+  }
+
+  Widget _buildChartRow(double maxWidth, double maxValue, T entity,
+      Series<T> series, int value, Color color)
+  {
+    if (maxValue <= 0) {
+      _log.w("Can't create a chart row with maxValue = 0");
+      return Empty();
+    }
+
+    return InkWell(
+      onTap: () => widget.onTapRow?.call(entity, series.displayDateRange.value),
```

Comment:
Should be set to null if widget.onTapRow is null. This will disable the InkWell animation.